notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,498 @@
+
+
+```python
+# HIDDEN
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+%matplotlib inline
+import ipywidgets as widgets
+from ipywidgets import interact, interactive, fixed, interact_manual
+import nbinteract as nbi
+
+from sklearn import linear_model as lm
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import KFold
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import PolynomialFeatures
+
+sns.set()
+sns.set_context('talk')
+np.set_printoptions(threshold=20, precision=2, suppress=True)
+pd.options.display.max_rows = 7
+pd.options.display.max_columns = 8
+```
+
+
+```python
+# HIDDEN
+def df_interact(df, nrows=7, ncols=7):
+    '''
+    Outputs sliders that show rows and columns of df
+    '''
+    def peek(row=0, col=0):
+        return df.iloc[row:row + nrows, col:col + ncols]
+    if len(df.columns) <= ncols:
+        interact(peek, row=(0, len(df) - nrows, nrows), col=fixed(0))
+    else:
+        interact(peek,
+                 row=(0, len(df) - nrows, nrows),
+                 col=(0, len(df.columns) - ncols))
+    print('({} rows, {} columns) total'.format(df.shape[0], df.shape[1]))
+```
+
+## Cross-Validation
+
+In the previous chapter, we observed that we needed a more accurate way of simulating the test error to manage the bias-variance trade off. Cross-validation provides a method of estimating our model error using a single observed dataset by separating data used for training from the data used for model selection and final accuracy. 
+
+## Train-Validation-Test Split
+
+One way to accomplish this is to split the original dataset into three disjoint subsets:
+
+* Training set: The data used to fit the model.
+* Validation set: The data used to select features.
+* Test set: The data used to report the model's final accuracy.
+
+After splitting, we select a set of features and a model based on the following procedure:
+
+1. For each potential set of features, fit a model using the training set. The error of a model on the training set is its *training error*.
+1. Check the error of each model on the validation set: its *validation error*. Select the model that achieves the lowest validation error. This is the final choice of features and model.
+1. Calculate the *test error*, error of the final model on the test set. This is the final reported accuracy of the model. We are forbidden from adjusting the features or model to decrease test error; doing so effectively converts the test set into a validation set. Instead, we must collect a new test set after making further changes to the features or the model.
+
+This process allows us to more accurately determine the model to use than using the training error alone.
```

Comment:
Let's add a bit more explanation here since we make a claim that isn't super obvious. For example, we could mention that the validation set simulates the test set since we didn't use it for model training.

Reply:
fixed